### PR TITLE
add UrBackup

### PR DESCRIPTION
**Pull request by josef109**
Add Urbackup support V2

### Checklist
- [x] Build rule `all-supported` completed successfully
- [ ] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

File: spk/urbackup/src/wizard/install_uifile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[{
+    "step_title": "Urbackup",
+    "items": [{
+	"type": "textfield",
+        "desc": "Urbackup backup location. At this location client backup files are saved",
+        "subitems": [{
+            "key": "wizard_backup_dir",
+            "desc": "Backup path",
+            "defaultValue": "/mnt/backups/urbackup",
```
Comment (ymartin59) [inline spk/urbackup/src/wizard/install_uifile:9]: I would propose /volume1/urbackup as default location on DSM. FAQ would invite to create a Share Folder named "urbackup". Advanced users may select an other volume according to their own setup.

File: spk/urbackup/Makefile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+DEFAULT_BACKUP_FOLDER = $(STAGING_DIR)/var/backup
+
+POST_STRIP_TARGET = urbackup_extra_install
+
+include ../../mk/spksrc.spk.mk
+
+.PHONY: urbackup_extra_install
+urbackup_extra_install:
+	install -m 755 -d $(STAGING_DIR)/var
+	@mkdir -p $(STAGING_DIR)/var/urbackup
```
Comment (ymartin59) [inline spk/urbackup/Makefile:51]: As makefile does not include anything in these directories in package itself, I would create folder structure in post installation function.

File: spk/urbackup/Makefile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+SERVICE_PORT_TITLE = $(DISPLAY_NAME)
+
+SERVICE_COMMAND = $${SYNOPKG_PKGDEST}/bin/urbackupsrv run -d -v error -u sc-urbackup --pidfile $${PID_FILE} --logfile $${LOG_FILE}
+
+
+# Admin link for in DSM UI
+ADMIN_PORT = $(SERVICE_PORT)
+
+
+INSTALL_PREFIX = /usr/local/$(SPK_NAME)
```
Comment (ymartin59) [inline spk/urbackup/Makefile:41]: This line should be removed as already defaults.

File: cross/urbackup/Makefile [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+COMMENT  = UrBackup is an easy to setup open source client/server backup system, that through a combination of image and file backups accomplishes both data safety and a fast restoration time.
+LICENSE  = GNU GPL
+
+CONFIGURE_ARGS = --enable-packaging --enable-embedded-cryptopp --with-zlib=$(STAGING_INSTALL_PREFIX) #--without-crypto --without-zlib --without-mail 
+GNU_CONFIGURE = 1
+ADDITIONAL_CXXFLAGS = -Os
+ADDITIONAL_CFLAGS = -Os
+
+
+INSTALL_TARGET = myInstall
```
Comment (ymartin59) [inline cross/urbackup/Makefile:20]: If this target is really necessary, please rename to match our naming policy: `urbackup_install`

File: spk/urbackup/src/wizard/install_uifile (hunk 1 above)
Comment (josef109) [reply spk/urbackup/src/wizard/install_uifile]: ok

File: spk/urbackup/Makefile (hunk 2 above)
Comment (josef109) [reply spk/urbackup/Makefile]: changed

File: cross/urbackup/Makefile (hunk 4 above)
Comment (josef109) [reply cross/urbackup/Makefile]: changed

**Comment by hgy59** [comment]
@josef109 Hi, I just rebased you branch to current head@master. Let's accomplish your good work!

I removed the commit 32ebf660ef8b93552e465c4fe275d9da10f4033a as I think it is not needed (and the descriptions was not clear about what is fixed)

**Comment by hgy59** [comment]
@josef109 anything we should consider for this package?

**Pull request closed without merging**